### PR TITLE
DEV: Use new setupHashtagAutocomplete + registerHashtagSearchParam APIs

### PR DESCRIPTION
**Pull request by martin-brennan**
Include context in the setupHashtagAutocomplete call, and use registerHashtagSearchParam to set up category + tag search for the chat composer. We cannot add channel search yet because we need to be able to look up channels by slugs first and also choose/render selected channels. More core work still to come.

c.f. https://github.com/discourse/discourse/pull/18592

**Do not merge until the core PR has merged!**

### CODE REVIEW

**Comment by jjaffeux** [comment]
I assume failures are due to core not being merged?

**Comment by jjaffeux** [comment]
I think we should be able to test a test rather easily for this no? No need to go crazy, just a single one to ensure the channel autocomplete is showing.

**Comment by martin-brennan** [comment]
Sure I will add a test @jjaffeux and yes I think it's core but will double check the tests locally.

File: assets/javascripts/discourse/initializers/chat-setup.js [hunk 1]
```diff
@@ -43,6 +43,11 @@ export default {
         });
       }
 
+      if (this.siteSettings.enable_experimental_hashtag_autocomplete) {
+        api.registerHashtagSearchParam("category", "chat-composer", 100);
+        api.registerHashtagSearchParam("tag", "chat-composer", 50);
+      }
```
Comment (lis2) [inline assets/javascripts/discourse/initializers/chat-setup.js:49]: Related to https://github.com/discourse/discourse/pull/18592/files#r998915006

Do you think we should use separate api block like `withPluginApi("0.14.1"`

Comment (martin-brennan) [reply assets/javascripts/discourse/initializers/chat-setup.js:49]: I don’t want to bump it yet since it’s experimental I’ll do that once it’s more solid

**Review by lis2 — APPROVED** [review]

File: test/javascripts/acceptance/composer-hashtag-autocomplete-test.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import { setCaretPosition } from "discourse/lib/utilities";
```
Comment (jjaffeux) [inline test/javascripts/acceptance/composer-hashtag-autocomplete-test.js:1]: great thanks

**Pull request merged**